src/core_plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -558,7 +575,19 @@ function discoverController($scope, config, courier, $route, $window, Notifier,
   // TODO: On array fields, negating does not negate the combination, rather all terms
   $scope.filterQuery = function (field, values, operation) {
     $scope.indexPattern.popularizeField(field, 1);
-    filterManager.add(field, values, operation, $state.index);
+
```

Comment:
I've started to pull this logic into a new module in my [next PR](https://github.com/Bargs/kibana/blob/enhanceKuery/src/ui/public/query_manager/index.js). I just wasn't sure what form this new API should take at the time I was writing this and I didn't want to reuse the filter_manager since filter is such an overloaded and confusing term.